src/cacheServices/CacheTransactionService.ts

```diff
@@ -42,44 +46,61 @@ export const transferFilter = (transaction: Transaction): boolean => {
 };
 
 /**
- * Parses through list of transactions and casts them to TransferTransactions so we
+ * Parses through list of transactions and casts them to CacheTransferTransaction so we
  * can have access to important transfer details
  * @param {Transaction} transaction
- * @returns {TransferTransaction}
+ * @returns {CacheTransferTransaction}
  */
-export const mapTransfer = (transaction: Transaction): TransferTransaction => {
+export const mapTransfer = (transaction: Transaction): CacheTransferTransaction => {
+  let mosaics: Array<Mosaic> = [];
+  let xem: XEM = new XEM(1);
   if (transaction.type == TransactionTypes.TRANSFER) {
-    return transaction as TransferTransaction;
+    const transferTX = transaction as TransferTransaction;
+    if (transferTX.containsMosaics()) {
+      mosaics = transferTX.mosaics();
+    } else {
+      xem = xemDetails(transferTX);
+    }
+    return new CacheTransferTransaction(transferTX.recipient, xem, transferTX.timeWindow,
+      transferTX.version, transferTX.fee, transferTX.message, transferTX.signature, mosaics,
+      transferTX.signer, transferTX.getTransactionInfo()
+    );
   } else if (transaction.type == TransactionTypes.MULTISIG && (transaction as MultisigTransaction).otherTransaction.type == TransactionTypes.TRANSFER) {
-    return (transaction as MultisigTransaction).otherTransaction as TransferTransaction;
+    const transferTX = (transaction as MultisigTransaction).otherTransaction as TransferTransaction;
+    if (transferTX.containsMosaics()) {
+      mosaics = transferTX.mosaics();
+    } else {
+      xem = xemDetails(transferTX);
+    }
+    return new CacheTransferTransaction(transferTX.recipient, xem, transferTX.timeWindow,
+      transferTX.version, transferTX.fee, transferTX.message, transferTX.signature, mosaics,
+      transferTX.signer, transferTX.getTransactionInfo()
+    );
   }
-  throw new Error('Transaction does not contain TransferTransaction');
+  throw new Error("Transaction does not contain TransferTransaction");
 };
 
 /**
  * Verifies that mosaic received is cache and returns amount
  * @param {TransferTransaction} transaction
  * @returns {boolean}
  */
-export const cacheAmount = (transaction: TransferTransaction): number => {
+export const cacheDetails = (transaction: TransferTransaction): CACHE => {
   if (transaction.containsMosaics()) {
-    transaction.mosaics().map(mosaic => {
-      if (mosaic.mosaicId.namespaceId === 'cache' && mosaic.mosaicId.name === 'cache') {
-        return mosaic.quantity * transaction.xem().amount;
-      }
-    });
+    const cache = transaction.mosaics().find(mosaic => mosaic.mosaicId.namespaceId === CACHE.MOSAICID.namespaceId && mosaic.mosaicId.name === CACHE.MOSAICID.name);
+    if (cache) { return new CACHE(cache.quantity / Math.pow(10, CACHE.DIVISIBILITY)); }
```

Comment:
This code scares me a little bit. Can we extract it into a function so it can be re-used and unit tested ie
`cache.quantity / Math.pow(10, CACHE.DIVISIBILITY)`

Clearly this is a very specific math that we need to capture. Also it will help us be consistent because I have done this math different in the past using scientific notation.